### PR TITLE
Refactor API client

### PR DESCRIPTION
**Pull request by therve**
This moves thread support from the main client, and extract the behavior
flags in the configuration.

### CODE REVIEW

File: src/datadog_api_client/v1/api/authentication_api.py [hunk 1]
```diff
@@ -48,37 +48,6 @@ def validate(self, **kwargs):
...
-            one the data received from the server. Default is True.
-        :type _check_return_type: bool
-        :param _host_index: Specifies the index of the server that we want to
-            use. Default is read from the configuration.
-        :type _host_index: int/None
-        :param async_req: Execute request asynchronously.
-        :type async_req: bool
-
-        :return: If the method is called asynchronously, returns the request thread.
         :rtype: AuthenticationValidationResponse
```
Comment (jirikuncar) [inline src/datadog_api_client/v1/api/authentication_api.py:51]: This is not always true when using Threaded client.

File: .generator/src/generator/templates/api_client.j2 [hunk 2]
```diff
@@ -70,26 +63,11 @@ class ApiClient(object):
         self.close()
 
     def close(self):
-        if self._pool:
-            self._pool.close()
-            self._pool.join()
-            self._pool = None
-            if hasattr(atexit, "unregister"):
-                atexit.unregister(self.close)
+        pass
```
Comment (jirikuncar) [inline .generator/src/generator/templates/api_client.j2:66]: do we need this method at all?

Comment (therve) [reply .generator/src/generator/templates/api_client.j2:66]: I think it makes some sense to make it part of the interface? I need to look into closing the urllib pool manager as well.

Comment (therve) [reply .generator/src/generator/templates/api_client.j2]: Done, I added the clear we had in the tests.

File: README.md [hunk 3]
```diff
@@ -85,6 +85,23 @@ If you want to enable requests logging, set the `debug` flag on your configurati
...
+
+```python
+from datadog_api_client import Configuration, ThreadedApiClient
+from datadog_api_client.v1.api import dashboards_api
+
+configuration = Configuration()
+with ThreadedApiClient(configuration) as api_client:
+    api_instance = dashboards_api.DashboardsApi(api_client)
+    result = api_instance.list_dashboards()
+    dashbooards = result.get()
```
Comment (NouemanKHAL) [inline README.md:101]:
```suggestion
    dashboards = result.get()
```

File: README.md [hunk 4]
```diff
@@ -85,6 +85,23 @@ If you want to enable requests logging, set the `debug` flag on your configurati
...
+    print(dashbooards)
```
Comment (NouemanKHAL) [inline README.md:102]:
```suggestion
    print(dashboards)
```

**Review by NouemanKHAL — APPROVED** [review]

**Pull request merged**